enhancements/20250107-cpu-memory-hotplug.md

```diff
@@ -0,0 +1,97 @@
+# CPU / Memory Hotplug
+
+## Summary
+
+This enhancement focuses on leveraging KubeVirt's CPU and Memory hotplug feature to enable users to dynamically adjust the CPU and Memory resources of a VM.
+
+### Related Issues
+
+https://github.com/harvester/harvester/issues/5835
+
+## Motivation
+
+### Goals
+
+- Users can adjust the CPU and Memory resources of a VM dynamically.
+- Users can set maximum CPU and Memory resources for a VM. The dynamic adjustment of CPU and Memory resources will be limited by the maximum resources.
+
+### Non-goals
+
+N/A
+
+## Proposal
+
+The cpu / memory hotplug feature will be enabled by default. When creating a VM, users can set 4 fields: initial cpu / memory and maximum cpu / memory. The initial cpu / memory will be the initial resources allocated to the VM. The maximum cpu / memory will be the maximum resources that the VM can be adjusted to.
+
+When the VM is running, users can adjust the CPU and Memory resources of the VM. The adjustment will be limited by the maximum resources.
+
+### User Stories
+
+#### Story 1 - Vertical Scaling a VM without downtime
+
+There is a legacy workload which is not actively developed. The workload was not designed as a distributed system, so it's hard to scale out. Recently, there are some users using this application, and the workload is getting heavier. The current resources allocated to the VM are not enough to handle the workload. The administration wants to adjust the CPU and Memory resources of the VM without downtime.
+
+### User Experience In Detail
+
+1. Prepare an Image.
+
+2. On the VM creation dashboard, click the `Enable CPU / Memory Hotplug` button and the dashboard will show another two fields: Maximum CPU and Maximum Memory.
+
+3. Fulfill the VM creation form and click the `Create` button.
+
+4. After the VM is created, administrators can adjust the CPU and Memory resources by clicking `CPU / Memory Hotplug` on the VM details page.
+
+### API changes
+
+#### Update CPU and Memomory resources of a VM [POST]
+
+Endpoint:  `/v1/harvester/vm/<vm>?action=cpuAndMemoryHotplug`
+
+Request (application/json):
+```json
+{
+  "sockets": 2,
+  "memory": "4Gi"
+}
+```
+
+## Design
+
+### Implementation Overview
+
+#### Initial CPU and Memory
```

Comment:
Is `initial cpu and memroy` similar to existing `requests`?

As `cpu and memory overcommitting` is widely used, can use still get such flexibility via setting a small `init cpu and memory` & big `max cpu and memory`?